app/Community/Actions/SendDailyDigestAction.php

```
@@ -0,0 +1,119 @@
+            }
+            $ids[$delayedSubscription->subject_type->value][] = $delayedSubscription->subject_id;
+        }
+        // and preload the titles for the associated records
+        $titles = [];
+        foreach ($ids as $type => $typeIds) {
+            $titles[$type] = match ($type) {
+                SubscriptionSubjectType::ForumTopic->value => ForumTopic::whereIn('id', $typeIds)->pluck('title', 'id'),
+                default => [],
+            };
+        }
+
+        // build the data to pass to the mail script
+        $notificationItems = [];
+        foreach ($delayedSubscriptions as $delayedSubscription) {
+            // if all the new posts have been deleted or aren't visible, ignore it
+            $handler = $this->getHandler($delayedSubscription->subject_type);
+            $count = $handler->getUpdatesSince($delayedSubscription);
+            if ($count > 0) {
+                $notificationItems[] = [
+                    'type' => $delayedSubscription->subject_type->value,
+                    'title' => $titles[$delayedSubscription->subject_type->value][$delayedSubscription->subject_id] ?? '(untitled)',
+                    'link' => $handler->getLink($delayedSubscription->subject_id, $delayedSubscription->first_update_id),
+                    'count' => $count,
+                ];
+            }
+        }
+
+        // send the mail
+        Mail::to($user->EmailAddress)->queue(
+            new DailyDigestMail($user, $notificationItems)
+        );
+    }
+
+    private function getHandler(SubscriptionSubjectType $subjectType): BaseDelayedSubscriptionHandler
+    {
+        return match ($subjectType) {
+            SubscriptionSubjectType::ForumTopic => new ForumTopicDelayedSubscriptionHandler(),
+
+            // this is wrong, but we have to return something to satisfy phpstan.
+            // other cases will be filled in as the calls are updated.
+            default => new ForumTopicDelayedSubscriptionHandler(),
```

Maybe it'd be better to throw an exception here?
```php
default => throw new InvalidArgumentException("Supported subject type: {$subjectType->value}"),
```